### PR TITLE
net-dns/knot: add 3.4.2

### PR DESCRIPTION
**Pull request by PPN-SD**
new module : [authsignal](https://gitlab.nic.cz/knot/knot-dns/-/commit/b37888237)

useflag :
dbus : useflag added to support [libdbus including when systemd is installed](https://gitlab.nic.cz/knot/knot-dns/-/commit/2d03ae85)
+doc : useflag enabled by default because pregenerated manpages are no longer provided by upstream
idn1 : useflag removed, [deprecated](https://gitlab.nic.cz/knot/knot-dns/-/commit/c379227)
[pkcs#11](https://www.knot-dns.cz/docs/3.4/html/appendices.html) : useflag added

scripts :
use upstream service for systemd
new openrc init script (knot-1.init) : 
- use background to create a root-owned pid file
- add checkconfig

use knot.tmpfile for permissions :
- /run/knot (pidfile created by knotd)
- /var/lib/knot (default database, template ...)

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by PPN-SD** [comment]
@Flowdalic, you are the last one to committed and I didn’t get a response from the maintainer with my old PR (3.4.0).
Do you think it is possible to review ?

**Comment by SoapGentoo** [comment]
can you update this to 3.4.2?

**Comment by PPN-SD** [comment]
> can you update this to 3.4.2?

I wait for the review of it first.
3.4.2 have no impact for ebuild. It will be a copy of this one.

**Comment by PPN-SD** [comment]
I’ll do it soon.

**Comment by PPN-SD** [comment]
@SoapGentoo updated to 3.4.2.
Remove version and slot deps for ngtcp2.
Do you think doc useflag by default with sphinx dep is a problem ?

**Comment by PPN-SD** [comment]
@SoapGentoo Do you think a common review with knot #39065 and knot-resolver #39069 is possible ?

I’m working on version knot-resolver-6.0 with dep >=knot-3.3 in parallel but I have difficulty seeing the future of this package.

File: net-dns/knot/metadata.xml [hunk 1]
```diff
@@ -10,9 +10,18 @@
...
+			Enable Automatic Authenticated DNSSEC Bootstrapping records
+		</flag>
 		<flag name="cookies">
 			Enable the DNS cookie module (RFC 7873)
 		</flag>
+		<flag name="daemon">
+			Enable Knot DNS main daemon
+		</flag>
+		<flag name="dbus">
+			Enable DBus support as an alternative of systemd d-bus
```
Comment (thesamesam) [inline net-dns/knot/metadata.xml:23]: I'm not sure if I follow this. Also, dbus and pkcs11 are global USE flags. You can provide a local definition here if it's useful but these two definitions don't seem to add much information either...

Comment (PPN-SD) [reply net-dns/knot/metadata.xml]: @thesamesam, it seems pkcs11 is not global : [use.decs](https://github.com/gentoo/gentoo/blob/master/profiles/use.desc#L262).

I deleted dbus.

**Comment by thesamesam** [comment]
Can you make the commit message descriptive (~matching the PR description)? Looks good to merge then.

**Comment by PPN-SD** [comment]
> Can you make the commit message descriptive (~matching the PR description)?

Done.

> Looks good to merge then.
Thanks to you.
This PR was a prerequisite for that one : #39482

**Comment by thesamesam** [comment]
Thanks! Merged with commit message slightly reformatted and some `Closes:` and `Bug:` tags. Please take that into account when rebasing the knot-resolver PR (go over its open bugs, tag them as appropriate), then when it is ready, I can take a look at that too.

**Pull request closed without merging**